pf/tfbridge/provider_create.go

```diff
@@ -44,6 +45,23 @@ func (p *provider) CreateWithContext(
 
 	priorState := newResourceState(ctx, &rh)
 
+	for key, fld := range rh.pulumiResourceInfo.Fields {
+		if _, ok := checkedInputs[resource.PropertyKey(key)]; !ok && fld.Default != nil {
+			// using default value for empty property
+			v, err := convert.GetDefaultValue(fld.Default, &tfbridge.PulumiResource{
+				URN:        urn,
+				Properties: checkedInputs,
+				Seed:       make([]byte, 0),
+			})
+			if err != nil {
+				return "", nil, 0, err
+			}
+			checkedInputs[resource.PropertyKey(key)] = resource.PropertyValue{
```

Comment:
There's an implicit conversion here from `any` aka `interface{}` to `resource.PropertyValue`. I think that's not quite right, I'd suggest this.

First cast and check if this is a resource.PropertyValue already, if so use that. This will let the users define precisely what they want if needed.

Otherwise fallback to `resource.NewPropertyValue` conversion. This will take care of converting slices, arrays and so on deeply into the `PropertyValue` representation.

Reply:
Implemented your proposal in the new `SetDefaultValues`